src/rgw/rgw_json_enc.cc

```diff
@@ -1995,14 +1995,14 @@ void lc_op::dump(Formatter *f) const
     f->dump_object("obj_tags", *obj_tags);
   }
   f->open_object_section("transitions");  
-  for(auto iter = transitions.begin(); iter != transitions.end(); iter++){
-    f->dump_object(iter->first.c_str(), iter->second);
+  for(auto& [storage_class, transition] : transitions){
+    f->dump_object(storage_class.c_str(), transition);
   }
   f->close_section();
 
   f->open_object_section("noncur_transitions");  
-  for(auto iter = noncur_transitions.begin(); iter != noncur_transitions.end(); iter++){
-    f->dump_object(iter->first.c_str(), iter->second);
+  for(auto& [storage_class, transition] : noncur_transitions){
+    f->dump_object(storage_class.c_str(), transition);
```

Comment:
```suggestion
    f->dump_object(storage_class, transition);
```

nit, since `dump_object()` accepts `string_view`, we can avoid the explicit call of `c_str()`. 

could you update the three places?

Reply:
sure, it's all taken care of.  Thanks for reviewing.